### PR TITLE
feat(input): add helper and error text

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: N/A

When using `ion-item` developers can provide optional helper and error text to assist with data entry.

## What is the new behavior?

- Adds functionality + styling for `helperText`
- Adds functionality + styling for `errorText`
- Adds styling for the "bottom content" (the area below the input/label that will house the helper text, error text, and counter)
- Adds the 'legacy-input' class to make it easier to detangle styles that should only apply to the legacy input.
- The error and helper slots on ion-item have been deprecated.

Note: counter logic will be added in a separate PR.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/input/test/bottom-content/input.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  test('should not render bottom content if no hint or counter is enabled', async ({ page }) => {
+    await page.setContent(`<ion-input label="my input"></ion-input>`);
+
+    const bottomEl = page.locator('ion-input .item-bottom');
+    await expect(bottomEl).toHaveCount(0);
+  });
+});
+
+test.describe('input: hint text', () => {
+  test.describe('input: hint text functionality', () => {
```
Comment (amandaejohnston) [inline core/src/components/input/test/bottom-content/input.e2e.ts:18]: I feel like these screenshot tests should include the whole input, not just the bottom part, to check things like position and padding between the two.

Comment (liamdebeasi) [reply core/src/components/input/test/bottom-content/input.e2e.ts:18]: The padding for the bottom text is set on the `.input-bottom` element which is captured in the screenshots. I can also add some screenshots to the `basic` test which checks the component as a whole. Would that work?

Comment (amandaejohnston) [reply core/src/components/input/test/bottom-content/input.e2e.ts:18]: Yeah, that sounds good 👍

Comment (liamdebeasi) [reply core/src/components/input/test/bottom-content/input.e2e.ts:18]: I'm going to add this in a separate PR with the `basic` tests that look at the component as a whole. Should be in https://github.com/ionic-team/ionic-framework/pull/26238 once I get everything ready.

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**